internal/healthchecks/healthchecks.go

```diff
@@ -20,13 +20,15 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+
+	"github.com/GoogleCloudPlatform/ops-agent/internal/logs"
 )
 
 var healthChecksLogFile = "health-checks.log"
 
 type HealthCheck interface {
 	Name() string
-	RunCheck(logger *log.Logger) error
+	RunCheck(logger *logs.FileLogger) error
 }
 
 type HealthCheckResult struct {
```

Comment:
Logging of Health Checks with multiple results (added in https://github.com/GoogleCloudPlatform/ops-agent/pull/1139) doesn't work well with the new FileLogger. It currently concatenates the results (see `func (r HealthCheckResult) String() string`) with a `\n` and it more or less worked (showed in separate lines) with the old default logger, but with the `FileLogger` we get : 

```
{"level":"info","time":"2023-05-03T20:49:57Z","caller":"healthchecks/healthchecks.go:102","message":"[Network Check] Result: FAIL, Error code: LogApiConnErr, Failure: Request to Logging API failed., Solution: Check your internet connection and firewall rules., Resource: https://cloud.google.com/logging/docs/agent/ops-agent/troubleshooting\n[Network Check] Result: FAIL, Error code: MonApiConnErr, Failure: Request to Monitoring API failed., Solution: Check your internet connection and firewall rules., Resource: https://cloud.google.com/monitoring/agent/ops-agent/troubleshooting\n[Network Check] Result: FAIL, Error code: PacApiConnErr, Failure: Request to packages.cloud.google.com failed., Solution: Check your internet connection and firewall rules., Resource: https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/troubleshoot-run-ingest#network-issues\n[Network Check] Result: FAIL, Error code: DLApiConnErr, Failure: Request to dl.google.com failed, Solution: Check your internet connection and firewall rules., Resource: https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/troubleshoot-run-ingest#network-issues","ops-agent-version":"2.31.0"}
```

I've been thinking in how to solve this. I propose the following : 
1. Create a method `ToStringSlice()` and use it to log from `RunAllHealthChecks` : 
```
func (r HealthCheckResult) ToStringSlice() []string {
	if mwErr, ok := r.Err.(MultiWrappedError); ok {
		var messageList []string
		for _, e := range mwErr.Unwrap() {
			messageList = append(messageList, singleErrorResultMessage(e, r.Name))
		}
		return messageList
	}
	return []string{singleErrorResultMessage(r.Err, r.Name)}
}
```

2. Rewrite the `String()` method like :
```
func (r HealthCheckResult) String() string {
	return strings.Join(r.ToStringSlice(), "\n")
}
```

Reply:
Thanks for the catch! Will look into this.